File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const test = [];
    for (let i = 0; i < priorities.length; i++) {
        test.push([i, priorities[i]]);
    }
    
    let count = 0;
    
    while(true) {
        const max = test.reduce((a, b) => a[1] > b[1] ? a : b)[1];
        
        let shift;
        if (test[0][1] < max) {
            const temp = test.shift();
            test.push(temp);
        } else if (test[0][1] === max) {
            count++;
            shift = test.shift();
        }
        
        if (shift && shift[0] === location) {
            break;
        }
    }
     return count;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue = deque(list([i, priorities[i]] for i in range(len(priorities))))
    result = 0
    
    while len(queue) > 1:
        wait_process = queue.popleft()
        max_process = max(queue, key=lambda x: x[1])
        
        if wait_process[1] < max_process[1]:
            queue.append(wait_process)
        else:
            result += 1
            if wait_process[0] == location:
                return result
    return result + 1